File: variants/java-maven2/net.pijcke.nim.aione/src/main/java/net/pijcke/nim/aione/OneAIFactory.java
package net.pijcke.nim.aione;

import net.pijcke.nim.rules.Player;
import net.pijcke.nim.rules.PlayerFactory;

public class OneAIFactory implements PlayerFactory {

    @Override
    public String label() {
        return "AI : One";
    }

    @Override
    public Player build() {
        return new OneAI();
    }
}

File: variants/kotlin-gradle/nim/aione/src/main/kotlin/net/pijcke/nim/aione/OneAiFactory.kt
package net.pijcke.nim.aione

import net.pijcke.nim.rules.Player
import net.pijcke.nim.rules.PlayerFactory

class OneAiFactory : PlayerFactory {

    override fun label() = "AI : One"

    override fun build(): Player {
        return OneAi()
    }

}